bigtable/snippets/src/main/java/com/example/bigtable/Reads.java

```diff
@@ -0,0 +1,216 @@
+// [START bigtable_reads_prefix]
+// [START bigtable_reads_filter]
+
+import static com.google.cloud.bigtable.data.v2.models.Filters.FILTERS;
+
+import com.google.api.gax.rpc.ServerStream;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.models.Filters;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.data.v2.models.RowCell;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.protobuf.ByteString;
+
+public class Reads {
+  // [END bigtable_reads_row]
+  // [END bigtable_reads_row_partial]
+  // [END bigtable_reads_rows]
+  // [END bigtable_reads_row_range]
+  // [END bigtable_reads_row_ranges]
+  // [END bigtable_reads_prefix]
+  // [END bigtable_reads_filter]
+
+  // [START bigtable_reads_row]
+  public static void readRow(String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+
+    try (BigtableDataClient dataClient = BigtableDataClient.create(projectId, instanceId)) {
+      String rowkey = "phone#4c410523#20190501";
+
+      Row row = dataClient.readRow(tableId, rowkey);
+      printRow(row);
+
+    } catch (Exception e) {
+      System.out.println("Error during readRow: \n" + e.toString());
+    }
+  }
+  // [END bigtable_reads_row]
+
+  // [START bigtable_reads_row_partial]
+  public static void readRowPartial(String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+
+    try (BigtableDataClient dataClient = BigtableDataClient.create(projectId, instanceId)) {
+      String rowkey = "phone#4c410523#20190501";
+      Filters.Filter filter =
+          FILTERS
+              .chain()
+              .filter(FILTERS.family().exactMatch("stats_summary"))
+              .filter(FILTERS.qualifier().exactMatch("os_build"));
+
+      Row row = dataClient.readRow(tableId, rowkey, filter);
+      printRow(row);
+
+    } catch (Exception e) {
+      System.out.println("Error during RowPartial: \n" + e.toString());
+    }
+  }
+  // [END bigtable_reads_row_partial]
+
+  // [START bigtable_reads_rows]
+  public static void readRows(String projectId, String instanceId, String tableId) {
+    // String projectId = "my-project-id";
+    // String instanceId = "my-instance-id";
+    // String tableId = "mobile-time-series";
+
+    try (BigtableDataClient dataClient = BigtableDataClient.create(projectId, instanceId)) {
+      Query query =
+          Query.create(tableId).rowKey("phone#4c410523#20190501").rowKey("phone#4c410523#20190502");
+      ServerStream<Row> rows = dataClient.readRows(query);
+      for (Row row : rows) {
+        printRow(row);
+      }
+
+    } catch (Exception e) {
```

Comment:
Please catch more specific errors - see [Exception Handling](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#exception-handling)

Reply:
(Same for other places)